src/steamship/agents/planner/react.py

```diff
@@ -87,12 +78,18 @@ def plan(self, context: AgentContext) -> Action:
             scratchpad=scratchpad,
         )
 
-        # print(f"Prompt: {prompt}\n----\n")
-        # print(f"LLM Scratchpad: {scratchpad}\n")
+        completions = self.llm.complete(prompt=prompt, stop="Observation:")
+        return self.output_parser.parse(completions[0].text, context)
 
-        gpt4 = context.client.use_plugin("gpt-4")
-        task = gpt4.generate(text=prompt, options={"stop": "Observation:"})
-        task.wait()
-        # here, we assume that the response will always be a single block of text.
-        # print(f"LLM output: {task.output.blocks} \n")
-        return self.output_parser.parse(task.output.blocks[0].text, context)
+    def _construct_scratchpad(self, context):
+        steps = []
+        for action in context.completed_steps:
+            steps.append(
+                "Thought: Do I need to use a tool? Yes"
```

Comment:
It would be so cool to start logging this in a structured way right form V1 -- that way we can visualize the thought process.

Reply:
Also do  we need \n chars here?